apps/studio/components/interfaces/Integrations/Vault/Secrets/DeleteSecretModal.tsx

```
@@ -1,22 +1,31 @@
-  onClose: () => void
-  onDeleteStart?: (secretId: string) => void
-}
+import { useVaultSecretsQuery } from '@/data/vault/vault-secrets-query'
 
-const DeleteSecretModal = ({ selectedSecret, onClose, onDeleteStart }: DeleteSecretModalProps) => {
+export const DeleteSecretModal = () => {
   const { data: project } = useSelectedProjectQuery()
-  const { mutate: deleteSecret, isPending: isDeleting } = useVaultSecretDeleteMutation({
+
+  const { data: secrets = [], isSuccess } = useVaultSecretsQuery({
+    projectRef: project?.ref!,
```

Biome (2.3.14)

[error] Forbidden non-null assertion after optional chaining.

Optional chaining already handles nullish values. Using non-null assertion defeats its purpose and may cause runtime errors.
Consider using the nullish coalescing operator `??` or optional chaining throughout the chain instead.

(lint/suspicious/noNonNullAssertedOptionalChain)